controllers/operator_controller.go

```diff
@@ -277,3 +289,21 @@ func (r *OperatorReconciler) existingBundleDeploymentUnstructured(ctx context.Co
 	}
 	return &unstructured.Unstructured{Object: unstrExistingBundleDeploymentObj}, nil
 }
+
+func (r *OperatorReconciler) verifySuccessfulBundleDeployments(ctx context.Context, name string) error {
+	bundleDeploymentExists := &rukpakv1alpha1.BundleDeployment{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: name}, bundleDeploymentExists); err != nil {
+		return fmt.Errorf("bundleDeployment %q not found %v", name, err)
```

Comment:
This could give us stale data since we've _just_ potentially patched the BD. We should use the BD updated via the `ensureBundleDeployment` function.

And I think we need to tweak that to handle the case where we don't actually make a patch call by making sure the desired BD is updated to point to the existing BD.

Reply:
Tweaked it a bit to check the existing and desired BD after patch.